app/kuma-dp/cmd/run.go

```diff
@@ -136,15 +136,27 @@ func newRunCmd(opts kuma_cmd.RunCmdOpts, rootCtx *RootContext) *cobra.Command {
 				cfg.Dataplane.Name = proxyResource.GetMeta().GetName()
 			}
 
-			if cfg.DataplaneRuntime.ConfigDir == "" || cfg.DNS.ConfigDir == "" {
+			if cfg.DataplaneRuntime.WorkDir == "" {
+				if cfg.DataplaneRuntime.ConfigDir != "" {
+					runLog.Info("configDir(--config-dir) in config file is deprecated, use workDir(--work-dir) instead")
+					cfg.DataplaneRuntime.WorkDir = cfg.DataplaneRuntime.ConfigDir
+				}
+				if cfg.DataplaneRuntime.SocketDir != "" {
+					runLog.Info("socketDir in config file is deprecated, use workDir(--work-dir) instead")
+				}
+			} else {
+				cfg.DataplaneRuntime.SocketDir = cfg.DataplaneRuntime.WorkDir
+			}
```

Comment:
The logic is incorrect. If `SocketDir` is explicitly set, it should take priority to avoid breaking existing setups. To handle this correctly, we need to differentiate between `SocketDir` and `WorkDir` instead of always overriding `SocketDir` based on `WorkDir`.

Wdyt:
```suggestion
			if cfg.DataplaneRuntime.WorkDir == "" && cfg.DataplaneRuntime.ConfigDir != "" {
				runLog.Info("ConfigDir is deprecated, please use WorkDir instead")
				cfg.DataplaneRuntime.WorkDir = cfg.DataplaneRuntime.ConfigDir
			}

			if cfg.DataplaneRuntime.SocketDir != "" {
				runLog.Info("SocketDir is deprecated, please use WorkDir instead")
			} else {
				cfg.DataplaneRuntime.SocketDir = cfg.DataplaneRuntime.WorkDir
			}
```